src/compiler/emitter.ts

```diff
@@ -1862,8 +1862,19 @@ namespace ts {
         }
 
         function emitModuleDeclaration(node: ModuleDeclaration) {
-            emitModifiers(node, node.modifiers);
-            write(node.flags & NodeFlags.Namespace ? "namespace " : "module ");
+            if (node.flags & NodeFlags.GlobalAugmentation) {
+                if (!hasModifier(node, ModifierFlags.Ambient)) {
+                    // Always emit a 'declare' keyword in case it wasn't provided by a factory function call.

```

Comment:
Maybe we should just throw an error when factory functions get invalid data? What's the protocol for something like this @rbuckton ?

Reply:
I think throwing an error is better, or changing the type to require the declare keyword.

Reply:
we do not do this in other places, why not just emit modifiers?

Reply:
`declare` isn't always legal, such as when meeting a global augmentation in a module augmentation:
```ts
// assuming there is a "foo" module...
declare module "foo" {
  global {
  }
}
```

Reply:
That is a good point, and @rbuckton I didn't realize that was even legal.